### PR TITLE
[promptflow] Increase `waitress-serve` number of threads

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

image

This PR targets to increase the number of the threads that `waitress-serve` uses, so that local PFS can serve better, especially when collecting huge amount of traces.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_constants.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def _prepare_home_dir() -> Path:
 PF_SERVICE_LOG_FILE = "pfs.log"
 PF_SERVICE_HOUR_TIMEOUT = 1
 PF_SERVICE_MONITOR_SECOND = 60
+PF_SERVICE_WAITRESS_THREADS = 16
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_constants.py:80]: maybe make this not mention thread, waitress is based on thread, but other WSGI host like gunicorn is process based.
Does this sounds better?
PF_SERVICE_WORKER_COUNT

Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_constants.py:80]: maybe a field in pf config ?

File: src/promptflow/promptflow/_sdk/_service/entry.py [hunk 2]
```diff
@@ -131,7 +131,10 @@ def validate_port(port, force_start):
                     f"Please install pywin32 by 'pip install pywin32' and retry. prompt flow "
                     f"service start depends on pywin32.. {ex}"
                 )
-            command = f"waitress-serve --listen=127.0.0.1:{port} promptflow._sdk._service.entry:get_app"
+            command = (
+                f"waitress-serve --listen=127.0.0.1:{port} --threads={PF_SERVICE_WAITRESS_THREADS} "
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/entry.py:135]: Any perf number with this?

File: src/promptflow/promptflow/_sdk/_constants.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_constants.py:80]: yes, maybe worker here is better

Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_constants.py:80]: config can be another PR? this just work like a default value

File: src/promptflow/promptflow/_sdk/_service/entry.py (hunk 2 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_service/entry.py:135]: not yet, Heyi told me that when he runs flow with many traces, the PFS will get stuck; as waitress uses thread, it should be fine with a little more thread worker, so this PR is kind of a workaround for current PFS stuck issue.

File: src/promptflow/promptflow/_sdk/_constants.py (hunk 1 above)
Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_constants.py]: updated to `PF_SERVICE_WORKER_NUM`

**Review by wangchao1230 — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**